process/block/metablock.go

```diff
@@ -1521,6 +1569,13 @@ func (mp *metaProcessor) ApplyBodyToHeader(hdr data.HeaderHandler, bodyHandler d
 		return nil, process.ErrWrongTypeAssertion
 	}
 
+	if check.IfNil(bodyHandler) {
+		return nil, process.ErrNilBlockBody
+	}
+	if check.IfNil(mp.blockChain) {
```

Comment:
This check could be removed as it is checked for nil on constructor

Reply:
done.